tesseract-server/src/handlers/metadata.rs

```diff
@@ -44,14 +44,26 @@ pub fn metadata_all_handler(
     ) -> ActixResult<HttpResponse>
 {
     info!("Metadata for all");
-    let mut schema_details = req.state().schema.read().unwrap().metadata();
+    let user_auth_level = get_user_auth_level(&req);
+    let mut schema_details = req.state().schema.read().unwrap().metadata(user_auth_level);
     let ll_config = match &req.state().logic_layer_config {
         Some(llc) => llc.read().unwrap().clone(),
-        None => return  Ok(HttpResponse::Ok().json(schema_details))
+        None => {
+            return  Ok(HttpResponse::Ok().json(schema_details))
+        }
     };
     let mut cubes: Vec<CubeMetadata> = Vec::new();
     for cube in schema_details.cubes.iter(){
-        cubes.push(get_cube_metadata(cube.clone(), &ll_config));
+        // Filter out cube that user isn't authorized to see
+        match user_auth_level {
+            Some(auth_level) => { // Authorization is set
+                if (auth_level >= cube.min_auth_level && auth_level >= 0) {
+                    cubes.push(get_cube_metadata(cube.clone(), &ll_config));
+                }
+            },
+            // No authorization set. Show all cubes
+            None => cubes.push(get_cube_metadata(cube.clone(), &ll_config))
```

Comment:
Similar to what I mentioned above, I think that when no level is provided, only the cubes with min_level = 0 should be returned.

Reply:
this is a case though where the server is not using JWT _at all_ ... because we don't want to necessarily force it. In case the JWT is set but auth is invalid it will be Some(-1)